egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list_file is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list_file):
+        raise Exception(args.rir_list_file + "not found")
+    
+    if args.noise_list_file is not None:
+        if not os.path.isfile(args.noise_list_file):
+            raise Exception(args.noise_list_file + "not found")
+
+    if args.num_replica > 1 and args.prefix is None:
+        args.prefix = "rvb"
+        warnings.warn("--prefix is set to 'rvb' as --num-replications is larger than 1.")
+
+    return args
+
+def ParseFileToDict(file, assert2fields = False, value_processor = None):
+    if value_processor is None:
+        value_processor = lambda x: x[0]
+
+    dict = {}
+    for line in open(file, 'r'):
+        parts = line.split()
+        if assert2fields:
+            assert(len(parts) == 2)
+
+        dict[parts[0]] = value_processor(parts[1:])
+    return dict
+
+
+# This is the major function to generate pipeline command for the corruption
+# The rir list would have the following format:
+# --rir-id <string,compulsary> --room-id <string,compulsary> --receiver-position-id <string,optional> --source-position-id <string,optional> --rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >
+# The noise list would have the following format:
+# --noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> --bg-fg-type <choices=(background|foreground), default=background> --rir-file <str, compulsary if isotropic, should not be specified if point-source> < location=(support Kaldi IO strings) >
+def CorruptWav(wav_scp, durations, output_dir, room_list, noise_list, snr_string, num_replica, prefix, speech_rvb_probability, noise_adding_probability, max_noises_added):
+    rooms = list_cyclic_iterator(room_list, random_seed = 1)
+    noises = None
+    if len(noise_list) > 0:
+        noises = list_cyclic_iterator(noise_list, random_seed = 1)
+    snrs = list_cyclic_iterator(snr_string.split(':'))
+    command_list = []
+    for i in range(num_replica):
+        keys = wav_scp.keys()
+        keys.sort()
+        for wav_id in keys:
+            wav_pipe = wav_scp[wav_id]
+            wav_dur = durations[wav_id]
+            if prefix is not None:
+                wav_id = prefix + str(i) + "_" + wav_id
+            command = "{0} {1} wav-reverberate".format(wav_id, wav_pipe)
+
+            # pick the room
+            room = rooms.next()
+            command_opts = ""
+            noises_added = []
+            snrs_added = []
+            start_times_added = []
```

Comment:
I think there should be another for durations of noise added. For isotropic, you can make it the whole file. For others, you have to choose randomly. We could probably have choose some random duration < max-duration. It might be better to make it separate paramters for background and foregroud noises because the foreground noises are short and you don't want to make the duration too long. 
@vijayaditya Do you have any suggestions regarding how to choose duration?